iceoryx_utils/test/moduletests/test_cxx_vector.cpp

```diff
@@ -1116,3 +1116,99 @@ TEST_F(vector_test, PopBackReturnsTrueTillItsEmpty)
 
     EXPECT_FALSE(sut.pop_back());
 }
+
+TEST_F(vector_test, ResizeFailsWhenCountIsGreaterThanCapacity)
+{
+    EXPECT_FALSE(sut.resize(sut.capacity() + 1));
+}
+
+TEST_F(vector_test, ResizeWithTemplateValueFailsWhenCountIsGreaterThanCapacity)
+{
+    EXPECT_FALSE(sut.resize(sut.capacity() + 1, 12));
+}
+
+TEST_F(vector_test, SizeIncreaseWithResizeAndDefaultCTorWorks)
+{
+    class DefaultCTor
+    {
+      public:
+        DefaultCTor()
+            : m_a{1231}
+        {
+        }
+        int m_a;
+    };
+    iox::cxx::vector<DefaultCTor, 10> sut;
+
+    EXPECT_TRUE(sut.resize(5));
+    for (auto& e : sut)
+    {
+        EXPECT_THAT(e.m_a, Eq(1231));
+    }
+    EXPECT_THAT(sut.size(), Eq(5));
```

Comment:
Maybe this should be checked before the for loop (and replaced by ASSERT_*). Consider this also for the following tests.